src/blog/better-class-features.md

```diff
@@ -0,0 +1,332 @@
+
+```js
+class A extends class {} {
+  #method();
+  constructor(run) {
+    const callSuper = () => super();
+    // ...do something
+    run(callSuper)
+  }
+};
+
+new A((fn) => fn());
+```
+
+V8 now generates the following bytecode:
+
+```cpp
+// Invoke the super constructor to construct the instance
+// and store it into r3.
+...
+
+// Load the private brand symbol from the class context at
+// depth 1 from the current context and store it into r4
+LdaImmutableContextSlot <context>, [3], [1]
+Star4
+
+// Load the depth 1 as an Smi into r6
+LdaSmi [1]
+Star6
+
+// Load the current context into r5
+Mov <context>, r5
+
+// Use the %AddPrivateBrand() to locate the class context at
+// depth 1 from the current context and store it in the instance
+// with the private brand symbol as key
+CallRuntime [AddPrivateBrand], r3-r6
+```
+
+In this case the cost of the runtime call is back so initializing instances of this class is still going to be slower compared to initializing instances of classes with only public methods. It is possible to use a dedicated bytecode to implement what `%AddPrivateBrand()` does, but since invoking `super()` in a nested arrow function is quite rare, we again traded the performance for simplicity of the implementation.
+
+## Final notes
+
+The work mentioned in this blog post is also included in the upcoming Node.js v18 release. Previously, Node.js switched to symbol properties in a few built-in classes that had been using private fields in order to include them into the embedded bootstrap snapshot as well as to improve the performance of the constructors (see [this blog post](https://www.nearform.com/blog/node-js-and-the-struggles-of-being-an-eventtarget/) for more context). With the improved support of class features in V8, Node.js [switched back to private class fields](https://github.com/nodejs/node/pull/42361) in these classes and Node.js's benchmarks showed that [these changes did not introduce any performance regressions](https://github.com/nodejs/node/pull/42361#issuecomment-1068961385).
+
+We'd like to thank Bloomberg for sponsoring Igalia to work on improvements of the class features and startup snapshots.
```

Comment:
```suggestion
This work was contributed to V8 by Igalia under the sponsorship of Bloomberg.
```
@lutzvahl Any concerns about my suggested wording and calling out who did the work?

Reply:
This looks like the only remaining unresolved comment with the blog post now - so pinging @lutzvahl, any comments about this?

Reply:
Lutz and I chatted internally. Let's go with the historically used phrasing "Thanks to Igalia and Bloomberg for contributing this implementation!"

Reply:
Done